### DISCUSSION TITLE
Lanchers not showing up in steam library

### DISCUSSION THREAD

**Discussion by Danither3al** [moraroy/NonSteamLaunchers-On-Steam-Deck#158] [Q&A]
I followed the guide but the launchers i need (Epic games and rockstar for GTA V) are not appearing on the library and i don't know if they are even being installed since i can't find them everywhere pls help asap.

**Comment by moraroy** [comment]
@Danither3al were you able to get this fixed? Make sure Proton GE is installed properly, download from the discover store Proton Up QT and then if you see Proton GE installed delete it. And then install the latest one again. The script is having trouble extracting the tar file.

**Reply by DOMMOMEISTER** [reply]
Hey can you help me, Ubisoft connect doesn't show up in my steam library after installing, I already made another thread about it on here, I'd appreciate the help

**Comment by Danither3al** [comment]
No i tried uninstallng using Proton Up QT and re insatllinf the latest proton but i get the same error : the program says it's downloading It but when steam restart itself the laucnhers aren't there

**Comment by Danither3al** [comment]
Update : i delete a file and now the laucnhers work

**Reply by DOMMOMEISTER** [reply]
Which file did you delete? I got the same problem with Ubi connect

**Comment by DOMMOMEISTER** [comment]
Hey man, which file did you delete to get it to show up in your library again?

**Comment by moraroy** [comment]
@Danither3al @DOMMOMEISTER I just updated to 3.2 hopefully now you won't have that problem!

**Reply by Danither3al** [reply]
Now it stopped working again

**Reply by moraroy** [reply]
@Danither3al send your full log file

**Comment by Hakoexe** [comment]
I am also facing this same issue!

**Reply by moraroy** [reply]
@Hakoexe  can you post your log?

**Reply by Hakoexe** [reply]
[NonSteamLaunchers-install.log](https://github.com/moraroy/NonSteamLaunchers-On-Steam-Deck/files/13976446/NonSteamLaunchers-install.log)


I believe these are my logs

**Reply by Hakoexe** [reply]
@moraroy Did you get everything you needed? Or find anything interesting

**Reply by moraroy** [reply]
@Hakoexe try it now, not sure if that fixed both of your errors though it may have justt fixed one of them though im still working on it

**Reply by Hakoexe** [reply]
Looks like it works for me! Thanks again for the help!

**Comment by PokeAnimeMatt** [comment]
Hello all.

Sadly I am facing this issue too, I have tried ProtonUP QT and did a Fresh Start but Ubisoft Connect is still not showing up in my Non Steam Library.

Here is my install log.
[NonSteamLaunchers-install.log](https://github.com/moraroy/NonSteamLaunchers-On-Steam-Deck/files/14228617/NonSteamLaunchers-install.log)

**Reply by moraroy** [reply]
@PokeAnimeMatt i just added a small fix just i would still again do this just in case Install Proton Up QT from the Discovery Store, from there delete the Proton GE's that you have and Reinstall the latest Proton GE, if you dont want to do that you can go here manually and delete any Proton GE folders you have, its possible the proton got corrupted somehow /home/deck/.steam/root/compatibilitytools.d/ Once you ve done that i would then install Proton GE through Proton UP QT and then use NonSteamLaunchers to ensure there is no corruption.

**Reply by PokeAnimeMatt** [reply]
Good evening

Thank you for the quick response, I reinstalled Proton Up QT and the latest Proton GE was installed so I removed it and checked the folders like you mentioned, nothing was in their, so I reinstalled the latest Proton GE through PUpQT and loaded up NSLauncher, installed Ubisoft Connect but sadly it is still not appearing in my Non Steam folder.

Now back in desktop mode looking at NSLauncher it does actually have a file and location next to Ubisoft Connect so I am assuming it did actually install correctly and for some reason I can't find it in Steam Gaming mode, non steam games.

My updated install log below.

[NonSteamLaunchers-install.log](https://github.com/moraroy/NonSteamLaunchers-On-Steam-Deck/files/14230925/NonSteamLaunchers-install.log)

**Reply by moraroy** [reply]
@PokeAnimeMatt are you using the decky plugin or the .desktop file?

**Reply by PokeAnimeMatt** [reply]
The .desktop file from this GitHub page.

**Reply by moraroy** [reply]
@PokeAnimeMatt i put another fix in...let me know how it goes try everything from scratch again then let me know

**Reply by PokeAnimeMatt** [reply]
I'm afraid it's still the same, I did a Fresh Start, fresh download of the nonsteamlauncher, uninstalled proton and reinstalled a d even reverted back to SteamOs Stable from preview.

I do appreciate all your help and effort so far. Thank you.

[NonSteamLaunchers-install.log](https://github.com/moraroy/NonSteamLaunchers-On-Steam-Deck/files/14231028/NonSteamLaunchers-install.log)

**Reply by moraroy** [reply]
@PokeAnimeMatt do me a huge favor delete the log file you just sent and send me a fresh one lol

**Reply by PokeAnimeMatt** [reply]
I'm sorry, I wasn't aware that file was not an updated log.

Here ya go, a fresh start and install of Ubisoft Connect below.

[NonSteamLaunchers-install.log](https://github.com/moraroy/NonSteamLaunchers-On-Steam-Deck/files/14231792/NonSteamLaunchers-install.log)

**Reply by moraroy** [reply]
@PokeAnimeMatt start fresh and run it again

**Reply by PokeAnimeMatt** [reply]
I have just restarted and done a fresh start of both Proton and NSLauncher and Ubisoft Connect is still not appearing in my NonSteam games :(

I believe I'm following all your steps, is it possible there is something on the Deck preventing Ubisoft Connect from visually appearing in the NonSteam Folder?

Because when I load up NonSteamLauncher next to Ubisoft Connect an arrow points and has a directory of the location of the install, so I believe the NonSteamLauncher has installed it but for some odd reason it is not appearing anywhere when I return to Gaming mode.


[NonSteamLaunchers-install.log](https://github.com/moraroy/NonSteamLaunchers-On-Steam-Deck/files/14231819/NonSteamLaunchers-install.log)

**Reply by moraroy** [reply]
@PokeAnimeMatt no don't worry it's an error on my part there was alot of code moved around this week

**Reply by PokeAnimeMatt** [reply]
I do apologies for getting in contact again but even with the multiple updates you hgave provided, including the latest update and with my Steam Deck being in Public and Preview build the Ubisoft Launcher is still not showing up in my Non Steam Game Library but in the NSGLauncher in Desktop mode Ubisoft+ looks to be installed.

Any suggestions on to how to resolve this?

**Reply by moraroy** [reply]
@PokeAnimeMatt send the log

**Reply by PokeAnimeMatt** [reply]
[NonSteamLaunchers-install.log](https://github.com/user-attachments/files/15846422/NonSteamLaunchers-install.log)

Thank you

**Comment by hihaoden** [comment]
[NonSteamLaunchers-install.log](https://github.com/moraroy/NonSteamLaunchers-On-Steam-Deck/files/14337048/NonSteamLaunchers-install.log)
aunchers dont show up on steam, I deleted files, even proton but only amazon and batle.net show up

**Comment by Danither3al** [comment]
This is still not working with latest release

**Comment by Danither3al** [comment]
Like the launcher are being downloaded but they don't create any shortcut to steam and I can't launch them

**Comment by moraroy** [comment]
please be patient as the Plugin Version of NSL fixes this problem mostly

**Comment by moraroy** [comment]
Make sure when you download the .Desktop file. You're dragging the .desktop file onto the desktop and Running it from there btw.

**Comment by Danither3al** [comment]
Still nothing

**Reply by moraroy** [reply]
@Danither3al have you tried the decky plugin? It gives rid of alot of vdf errors. Make sure there is no files in compatdata having to do with NSL and then go into deckys testing store and try that.

**Comment by The-original-Draco** [comment]
well, I kinda have a similar problem, when using the NonSteamLaunchers.desktop it adds everything twice to the Steam Library, NonSteamLaunchers, Epic Games as well as each Game, but when I use the NSLPlugin.desktop, it adds nothing to the Steam Library...

**Reply by R4Zi3L** [reply]
have the same problem

**Reply by moraroy** [reply]
@R4Zi3L I belive this to be fixed in the new version is this true?

**Comment by ninjalolacheese** [comment]
I'm having this issue too. I deleted Desktop version after It wouldn't add Ich.io. 

I installed the decky loader version which worked but I accidentally deleted compatdata from the decky storage cleaner plugin which broke the other launchers(EA App & GOG) & removed all the games. 

I tried using the fresh start option but it would loop indefinitely(Desktop ver. would delete everything immediately). After that I went to desktop to delete the the NSL file from compatdata & NSL launcher. I then reinstalled it which worked for EA & GOG but not Ich.io(it says installed but the dot is red). I don't think I uninstalled it properly. 

What do I do? Is there something I didn't delete? How long is the start fresh option supposed to take as I waited upwards of 15-20mins

I will have to give the log later. 

EDIT: I found a file for Ich.io in compatdata but I don't know what that means or if it contains the exe so I can add it manually.  Does this mean it's installed but it's just not adding it?

**Comment by BobbyDaFish** [comment]
I'm running into a few different issues trying to get this up and running.

First, I tried the CLI command to run the install script to add Epic Games, but ran into a strange issue where it was just repeatedly adding Moonlight and Epic to the library as new entries. After some light troubleshooting I decided to give the desktop application a go.

I did Start Fresh, and have removed all Proton GE installs and installed only the latest version. Still, when I run the application and select Epic Games, Battle.net, and Moonlight it seems to fail to install them. My steam library will repeatedly act as if a new item is appearing, then being removed once every 15 seconds or so. This behavior only stops when I reboot.

I've attached my install log, looks like it's stopping after it can't find the config.vdf file which does exist in the config folder as it should.

[NonSteamLaunchers-install.log](https://github.com/user-attachments/files/25349233/NonSteamLaunchers-install.log)

**Comment by noliat88** [comment]
Running into the same issue - EA and Ubi not adding to library. 
With EA, it only adds "Repair EA Play" to the desktop and steam library. 
Ubisoft Connect says downloading and installing, but don't see anything in library afterwards. Folder is created in compatdata, but didn't see the launcher executable in it.
Moonlight Streaming keeps getting added to both desktop and library, despite not selecting it.

**Reply by 100fino** [reply]
Same problem here with ubisoft and ea. I even checked nonsteamlauchers folder in compact data and no ea nor ubisoft folders have been inside. The installation probably fails and so nothing is installed.
